File: docs/site/samples/python/examples/guides_variants_discover.py
"""Discovering the selectable variants.

The ``variants`` accessor returns the declared variant map; read its keys to see
what is selectable (the default arm is not listed — it is the root body, name
``"default"``). Standalone program.
"""

from __future__ import annotations

from pathlib import Path

from prompting_press import Prompt

# The caller reads the definition; the library does no file I/O itself.
# Resolve the file next to this program (a real app uses its own path).
_HERE = Path(__file__).parent


def main() -> None:
    summary = Prompt.from_yaml((_HERE / "summary.yaml").read_text())

    assert sorted(summary.variants) == ["concise", "structured"]
    assert "concise" in summary.variants  # True


if __name__ == "__main__":
    main()

File: docs/site/samples/typescript/examples/guides_variants_discover.ts
/**
 * Discovering the selectable variants: `variants` is the declared variant map;
 * read its keys to see what is selectable (the default arm is not listed — it is
 * the root body, name `"default"`). Standalone program.
 */

import assert from "node:assert/strict";
import { readFileSync } from "node:fs";
import { fileURLToPath } from "node:url";
import { Prompt } from "prompting-press";

// The caller reads the definition; the library does no file I/O itself.
// Resolve the file next to this program (a real app uses its own path).
const defFile = (name: string) => fileURLToPath(new URL(name, import.meta.url));

const summary = Prompt.fromYaml(readFileSync(defFile("summary.yaml"), "utf8"));
const variants = summary.variants ?? {};

assert.deepEqual(Object.keys(variants).sort(), ["concise", "structured"]);
assert.ok("concise" in variants); // true
